app/models/manageiq/providers/azure/inventory/collector.rb

```diff
@@ -114,4 +137,19 @@ def download_template(uri)
     _log.error("Failed to download Azure template #{uri}. Reason: #{e.inspect}")
     nil
   end
+
+  # Do not use threads in test environment in order to avoid breaking specs.
+  #
+  def thread_limit
+    Rails.env.test? ? 0 : @thread_limit
+  end
+
+  # The point at which we decide to grab a full listing and filter internally
+  # instead of grabbing individual resources via parallel threads.
+  #
+  # The default is to resort to a single request for sets of 500 or less.
+  #
+  def record_limit(multiplier = 20)
+    @thread_limit * multiplier
+  end
```

Comment:
@Ladas Where is record_limit used?

Reply:
@bronaghs It will be used as a heuristic in the target collection code.

Reply:
In a next PR, I am actually extracting this to a setting

Reply:
Ah, I missed it above